### ISSUE TITLE
[1.5.5 -> master] unused-variable for dotted import

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/pylint#923]
I noticed a couple of issues when trying out astroid/pylint master, with a codebase which is pylint-clean with astroid 1.4.5 and pylint 1.5.5. I'll open an issue for thing I think is separate.

Those can all be reproduced by cloning the [qutebrowser repository](https://github.com/The-Compiler/qutebrowser) and running `tox -e pylint-master -r`. All line numbers are on commit 24db93f3eb17b97e8928d5faf26dd56221030493 of qutebrowser.

--- 

```
************* Module qutebrowser.misc.earlyinit
W:187, 8: Unused variable 'PyQt5' (unused-variable)
```

The error is a bit confusing as `PyQt5` just gets assigned as a consequence of `import PyQt5.QtCore`. *That* import is indeed unused, but I think the error should be suppressed as it's inside a `try`/`except ImportError`:

```python
def check_pyqt_core():
    """Check if PyQt core is installed."""
    try:
        import PyQt5.QtCore  # <-----
    except ImportError as e:
        # [show error]
        sys.exit(1)
```

**Comment by PCManticore** [comment]
Using the latest master, it shows correctly that PyQt5.QtCore is not used, maybe this got fixed, but I don't know when.
Anyway, do you think we shouldn't warn about unused imports if they are inside a TryExcept block with an ImportError? As far as I can tell right now, every other tool reports this import as unused, but this idiom might be used for determining if an app should run or do other things, as you're doing already in this snippet (and I'm using it as well in tests, for instance). If it seems a reasonable thing to not warn about, I guess I could change it (for instance, only if the except block doesn't contain an import, this would avoid false negatives, where a module was simply forgot).

**Comment by The-Compiler** [comment]
Hmm, I'm not actually sure - I guess complaining about it is okay, after all it's easily silenced.

**Comment by The-Compiler** [comment]
Also FWIW I still get:

```
************* Module qutebrowser.misc.earlyinit
W:187, 8: Unused variable 'PyQt5' (unused-variable)
```

**Comment by PCManticore** [comment]
Oh, I can reproduce it using that file, initially I was trying with that code snippet. Will look into it.
